repose-aggregator/components/filters/uri-stripper-filter/src/main/scala/org/openrepose/filters/uristripper/UriStripperFilter.scala

```diff
@@ -127,21 +124,81 @@ class UriStripperFilter @Inject()(configurationService: ConfigurationService)
         .fold(List.empty[LinkPath])(_ ++ _)
 
       if (token.isDefined && applicableLinkPaths.nonEmpty) {
-        val tryParsedJson = Try(Json.parse(wrappedResponse.getOutputStreamAsInputStream))
-        applicableLinkPaths.foldLeft(tryParsedJson)(transformLink(_, _, token.get, previousToken, nextToken)) match {
-          case Success(jsValue) =>
-            val jsValueBytes = jsValue.toString.getBytes(wrappedResponse.getCharacterEncoding)
-            wrappedResponse.setContentLength(jsValueBytes.length)
-            wrappedResponse.setOutput(new ByteArrayInputStream(jsValueBytes))
-          case Failure(e) =>
-            wrappedResponse.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR, e.getMessage)
+        Option(wrappedResponse.getContentType) match {
+          case Some(ct) if ct.toLowerCase.contains("json") => handleJsonResponseLinks(wrappedResponse, previousToken, nextToken, token, applicableLinkPaths)
+          case Some(ct) if ct.toLowerCase.contains("xml") => handleXmlResponseLinks(wrappedResponse, previousToken, nextToken, token, applicableLinkPaths)
+          case _ => //do nothing
         }
       }
 
       wrappedResponse.commitToResponse()
     }
   }
 
+  private def handleJsonResponseLinks(wrappedResponse: HttpServletResponseWrapper, previousToken: Option[String], nextToken: Option[String], token: Option[String], applicableLinkPaths: List[LinkPath]): Unit = {
+    val tryParsedJson = Try(Json.parse(wrappedResponse.getOutputStreamAsInputStream))
+    applicableLinkPaths.foldLeft(tryParsedJson)(transformJsonLink(_, _, token.get, previousToken, nextToken)) match {
+      case Success(jsValue) =>
+        val jsValueBytes = jsValue.toString.getBytes(wrappedResponse.getCharacterEncoding)
+        wrappedResponse.setContentLength(jsValueBytes.length)
+        wrappedResponse.setOutput(new ByteArrayInputStream(jsValueBytes))
+      case Failure(e) =>
+        wrappedResponse.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR, e.getMessage)
+    }
+  }
+
+  private def handleXmlResponseLinks(wrappedResponse: HttpServletResponseWrapper, previousToken: Option[String], nextToken: Option[String], token: Option[String], applicableLinkPaths: List[LinkPath]): Unit = {
+    try {
+      val result = applicableLinkPaths.foldLeft(wrappedResponse.getOutputStreamAsInputStream) { (in: InputStream, linkPath: LinkPath) =>
+        val out = new ByteArrayOutputStream()
+        val transformer = templateMap.get(linkPath).get.newTransformer
+        transformer.asInstanceOf[Controller].addLogErrorListener
+        transformer.setParameter("removedToken", token.getOrElse("")) //todo make optional
```

Comment:
I can remove that todo, it was from a while back. We agreed to pass an empty string instead and handle that logic in the xslt